src/NzbDrone.Common/Instrumentation/Sentry/SentryCleanser.cs

```diff
@@ -11,26 +11,41 @@ public static SentryEvent CleanseEvent(SentryEvent sentryEvent)
         {
             try
             {
-                sentryEvent.Message = CleanseLogMessage.Cleanse(sentryEvent.Message);
+                if (sentryEvent.Message is not null)
+                {
+                    sentryEvent.Message.Formatted = CleanseLogMessage.Cleanse(sentryEvent.Message.Formatted);
+                    sentryEvent.Message.Message = CleanseLogMessage.Cleanse(sentryEvent.Message.Message);
+                    sentryEvent.Message.Params = sentryEvent.Message.Params?.Select(x => CleanseLogMessage.Cleanse(x switch
```

Comment:
I've added the new fields because from the SDK version 2.0 to 3.0 the `Message` field moved from a single `string` to a complex type. So to keep it consistent I made the scrubbing run on all new fields.

I personally would remove all cleansing. Sentry itself has default PII stripping for things like password/auth/token/etc. And new rules can be added: https://docs.sentry.io/product/data-management-settings/scrubbing/

The SDK doesn't not add any PII willingly (unless `SendDefaultPii` it set to true explicitly).

If the goal is to remove data before it even reaches Sentry servers (Even though the connection to Sentry is via a TLS tunnel and data scrubbing happens before the data is stored in the database), one could host their own Relay and set their data scrubbing rules: https://docs.sentry.io/product/relay/#pii-data-scrubbing

Reply:
> If the goal is to remove data before it even reaches Sentry servers (Even though the connection to Sentry is via a TLS tunnel and data scrubbing happens before the data is stored in the database)

I believe this indeed is the goal. We ensure to users that error tracking stats are anonymous when they opt into this, so passing apikeys, tokens, etc. even to our private secured sentry instance is something we'd like to avoid.

Reply:
That makes sense. For the sake of this change I'd suggest we keep it as-is, with the current scrubbing done here (new one for consistency on new fields).
I'm happy to propose a new change that revisits the scrubbing approach and how this behaves with regards to Sentry and what data it can capture/drop.